airflow/api/common/experimental/mark_tasks.py

```diff
@@ -194,13 +196,15 @@ def get_subdag_runs(
 
             current_task = current_dag.get_task(task_id)
             if isinstance(current_task, SubDagOperator) or current_task.task_type == "SubDagOperator":
+                if TYPE_CHECKING:
+                    assert current_task.subdag
```

Comment:
Why this?

Reply:
Because mypy doesn't/can't know that `isinstance(current_task, SubDagOperator)` means `current_task.subdag` will be non-None as a result, so that's what this assert gives us.

Reply:
We could also probably do this _I think_ (not tested) -- as we have marked `SubDagOperator.subdag` of type `DAG`.

```python
current_task = cast(SubDagOperator, current_task)
```
or
```python
current_task.subdag = cast(DAG, current_task.subdag)
```

Reply:
The most generally accepted way to work around this is to actually use `assert` _without_ the if-guard; this is strictly for the linters, so `assert` being optimised in `-O` is totally a good thing. This is why Airflow’s “never use assert ever” rule has not sit well with me; assert exists for a reason, and _this is exactly it_.

Reply:
Well. What could I say :) . I was outvoted with that one :). So now there are the two of us.

Reply:
We can always ask for reassumption of the vote BTW if we have strong case.

Reply:
I'm more than happy to have the assert without the `if` -- I think the main point of avoiding asserts at runtime is that the code must continue to function if the assert was never run (and so for most people the easiest thing is to avoid them entirely)